src/Microsoft.IdentityModel.JsonWebTokens/JsonWebToken.cs

```diff
@@ -570,15 +570,17 @@ internal void ReadToken(ReadOnlyMemory<char> encodedTokenMemory)
         internal JsonClaimSet CreateClaimSet(ReadOnlySpan<char> strSpan, int startIndex, int length, bool createHeaderClaimSet)
         {
             int outputSize = Base64UrlEncoding.ValidateAndGetOutputSize(strSpan, startIndex, length);
+
             byte[] output = ArrayPool<byte>.Shared.Rent(outputSize);
             try
             {
-                Base64UrlEncoding.Decode(strSpan, startIndex, length, output);
+                ReadOnlySpan<char> slice = strSpan.Slice(startIndex, length);
```

Comment:
do we need the local variable slice?
Could we write: Base64UrlEncoder.UnsafeDecode(strSpan.Slice(startIndex, length), output);